Author: Jamie Strandboge <jamie@canonical.com>
Description: add membarrier syscall. This can be dropped once libseccomp is
 synced with 4.3 or higher
Forwarded: no

Index: libseccomp-2.2.3/src/arch-aarch64-syscalls.c
===================================================================
--- libseccomp-2.2.3.orig/src/arch-aarch64-syscalls.c
+++ libseccomp-2.2.3/src/arch-aarch64-syscalls.c
@@ -195,6 +195,7 @@ const struct arch_syscall_def aarch64_sy
 	{ "lstat", __PNR_lstat },
 	{ "lstat64", __PNR_lstat64 },
 	{ "madvise", 233 },
+	{ "mbarrier", 283 },
 	{ "mbind", 235 },
 	{ "memfd_create", 279 },
 	{ "migrate_pages", 238 },
Index: libseccomp-2.2.3/src/arch-arm-syscalls.c
===================================================================
--- libseccomp-2.2.3.orig/src/arch-arm-syscalls.c
+++ libseccomp-2.2.3/src/arch-arm-syscalls.c
@@ -207,6 +207,7 @@ const struct arch_syscall_def arm_syscal
 	{ "lstat", (__NR_SYSCALL_BASE + 107) },
 	{ "lstat64", (__NR_SYSCALL_BASE + 196) },
 	{ "madvise", (__NR_SYSCALL_BASE + 220) },
+	{ "mbarrier", (__NR_SYSCALL_BASE + 389) },
 	{ "mbind", (__NR_SYSCALL_BASE + 319) },
 	{ "memfd_create", (__NR_SYSCALL_BASE + 385) },
 	{ "migrate_pages", __PNR_migrate_pages },
Index: libseccomp-2.2.3/src/arch-mips64n32-syscalls.c
===================================================================
--- libseccomp-2.2.3.orig/src/arch-mips64n32-syscalls.c
+++ libseccomp-2.2.3/src/arch-mips64n32-syscalls.c
@@ -199,6 +199,7 @@ const struct arch_syscall_def mips64n32_
 	{ "lstat", (__NR_SYSCALL_BASE + 6) },
 	{ "lstat64", __PNR_lstat64 },
 	{ "madvise", (__NR_SYSCALL_BASE + 27) },
+	{ "mbarrier", __PNR_mbarrier },
 	{ "mbind", (__NR_SYSCALL_BASE + 231) },
 	{ "memfd_create", (__NR_SYSCALL_BASE + 318) },
 	{ "migrate_pages", (__NR_SYSCALL_BASE + 250) },
Index: libseccomp-2.2.3/src/arch-mips64-syscalls.c
===================================================================
--- libseccomp-2.2.3.orig/src/arch-mips64-syscalls.c
+++ libseccomp-2.2.3/src/arch-mips64-syscalls.c
@@ -199,6 +199,7 @@ const struct arch_syscall_def mips64_sys
 	{ "lstat", (__NR_SYSCALL_BASE + 6) },
 	{ "lstat64", __PNR_lstat64 },
 	{ "madvise", (__NR_SYSCALL_BASE + 27) },
+	{ "mbarrier", __PNR_mbarrier },
 	{ "mbind", (__NR_SYSCALL_BASE + 227) },
 	{ "memfd_create", (__NR_SYSCALL_BASE + 314) },
 	{ "migrate_pages", (__NR_SYSCALL_BASE + 246) },
Index: libseccomp-2.2.3/src/arch-mips-syscalls.c
===================================================================
--- libseccomp-2.2.3.orig/src/arch-mips-syscalls.c
+++ libseccomp-2.2.3/src/arch-mips-syscalls.c
@@ -199,6 +199,7 @@ const struct arch_syscall_def mips_sysca
 	{ "lstat", (__NR_SYSCALL_BASE + 107) },
 	{ "lstat64", (__NR_SYSCALL_BASE + 214) },
 	{ "madvise", (__NR_SYSCALL_BASE + 218) },
+	{ "mbarrier", __PNR_mbarrier },
 	{ "mbind", (__NR_SYSCALL_BASE + 268) },
 	{ "memfd_create", (__NR_SYSCALL_BASE + 354) },
 	{ "migrate_pages", (__NR_SYSCALL_BASE + 287) },
Index: libseccomp-2.2.3/src/arch-ppc64-syscalls.c
===================================================================
--- libseccomp-2.2.3.orig/src/arch-ppc64-syscalls.c
+++ libseccomp-2.2.3/src/arch-ppc64-syscalls.c
@@ -196,6 +196,7 @@ const struct arch_syscall_def ppc64_sysc
 	{ "lstat", 107 },
 	{ "lstat64", __PNR_lstat64 },
 	{ "madvise", 205 },
+	{ "mbarrier", 365 },
 	{ "mbind", 259 },
 	{ "memfd_create", 360 },
 	{ "migrate_pages", 258 },
Index: libseccomp-2.2.3/src/arch-ppc-syscalls.c
===================================================================
--- libseccomp-2.2.3.orig/src/arch-ppc-syscalls.c
+++ libseccomp-2.2.3/src/arch-ppc-syscalls.c
@@ -196,6 +196,7 @@ const struct arch_syscall_def ppc_syscal
 	{ "lstat", 107 },
 	{ "lstat64", 196 },
 	{ "madvise", 205 },
+	{ "mbarrier", 365 },
 	{ "mbind", 259 },
 	{ "memfd_create", 360 },
 	{ "migrate_pages", 258 },
Index: libseccomp-2.2.3/src/arch-s390-syscalls.c
===================================================================
--- libseccomp-2.2.3.orig/src/arch-s390-syscalls.c
+++ libseccomp-2.2.3/src/arch-s390-syscalls.c
@@ -179,6 +179,7 @@ const struct arch_syscall_def s390_sysca
 	{ "lstat", 107 },
 	{ "lstat64", 196 },
 	{ "madvise", 219 },
+	{ "mbarrier", __PNR_mbarrier },
 	{ "mbind", __PNR_mbind },
 	{ "memfd_create", 350 },
 	{ "migrate_pages", __PNR_migrate_pages},
Index: libseccomp-2.2.3/src/arch-s390x-syscalls.c
===================================================================
--- libseccomp-2.2.3.orig/src/arch-s390x-syscalls.c
+++ libseccomp-2.2.3/src/arch-s390x-syscalls.c
@@ -179,6 +179,7 @@ const struct arch_syscall_def s390x_sysc
 	{ "lstat", 107 },
 	{ "lstat64", __PNR_lstat64 },
 	{ "madvise", 219 },
+	{ "mbarrier", 356 },
 	{ "mbind", __PNR_mbind },
 	{ "memfd_create", 350 },
 	{ "migrate_pages", __PNR_migrate_pages },
Index: libseccomp-2.2.3/src/arch-x32-syscalls.c
===================================================================
--- libseccomp-2.2.3.orig/src/arch-x32-syscalls.c
+++ libseccomp-2.2.3/src/arch-x32-syscalls.c
@@ -195,6 +195,7 @@ const struct arch_syscall_def x32_syscal
 	{ "lstat", (X32_SYSCALL_BIT + 6) },
 	{ "lstat64", __PNR_lstat64 },
 	{ "madvise", (X32_SYSCALL_BIT + 28) },
+	{ "mbarrier", (X32_SYSCALL_BIT + 324) },
 	{ "mbind", (X32_SYSCALL_BIT + 237) },
 	{ "memfd_create", (X32_SYSCALL_BIT + 319) },
 	{ "migrate_pages", (X32_SYSCALL_BIT + 256) },
Index: libseccomp-2.2.3/src/arch-x86_64-syscalls.c
===================================================================
--- libseccomp-2.2.3.orig/src/arch-x86_64-syscalls.c
+++ libseccomp-2.2.3/src/arch-x86_64-syscalls.c
@@ -195,6 +195,7 @@ const struct arch_syscall_def x86_64_sys
 	{ "lstat", 6 },
 	{ "lstat64", __PNR_lstat64 },
 	{ "madvise", 28 },
+	{ "mbarrier", 324 },
 	{ "mbind", 237 },
 	{ "memfd_create", 319 },
 	{ "migrate_pages", 256 },
Index: libseccomp-2.2.3/src/arch-x86-syscalls.c
===================================================================
--- libseccomp-2.2.3.orig/src/arch-x86-syscalls.c
+++ libseccomp-2.2.3/src/arch-x86-syscalls.c
@@ -195,6 +195,7 @@ const struct arch_syscall_def x86_syscal
 	{ "lstat", 107 },
 	{ "lstat64", 196 },
 	{ "madvise", 219 },
+	{ "mbarrier", 375 },
 	{ "mbind", 274 },
 	{ "memfd_create", 356 },
 	{ "migrate_pages", 294 },
Index: libseccomp-2.2.3/include/seccomp.h.in
===================================================================
--- libseccomp-2.2.3.orig/include/seccomp.h.in
+++ libseccomp-2.2.3/include/seccomp.h.in
@@ -1557,6 +1557,11 @@ int seccomp_export_bpf(const scmp_filter
 #define __NR_s390_runtime_instr		__PNR_s390_runtime_instr
 #endif /* __NR_s390_runtime_instr */
 
+#define __PNR_mbarrier		-10197
+#ifndef __NR_mbarrier
+#define __NR_mbarrier		__PNR_mbarrier
+#endif /* __NR_mbarrier */
+
 #ifdef __cplusplus
 }
 #endif
